src/osd/OSD.cc

```diff
@@ -7714,6 +7802,14 @@ void OSD::consume_map()
 
       pg->unlock();
     }
+    for (auto pg = pending_creates_from_osd.cbegin();
+	 pg != pending_creates_from_osd.cend();) {
+      if (osdmap->get_pg_acting_rank(*pg, whoami) < 0) {
+	pg = pending_creates_from_osd.erase(pg);
+      } else {
+	++pg;
+      }
+    }
```

Comment:
if we removed something, we should probably call resume_creating_pg() here too?

Reply:
it has been taken care of: `_remove_pg()` is called in the `for` loop right after this block. and `_remove_pg()` calls `resume_creating_pg()`.